### ISSUE TITLE
issue with "runtime: failed to create new OS thread (have 2 already; errno=22) fatal error: newosproc" 

### ISSUE THREAD

**Issue by ninanung** [nginx-proxy/nginx-proxy#1550]
First, i'm using Apple Silicon M1 mac and i think this issue is related to it.
When i do `docker-compose up -d`, build is ok but container exited immediately.
```
WARNING: /etc/nginx/dhparam/dhparam.pem was not found. A pre-generated dhparam.pem will be used for now while a new one
is being generated in the background.  Once the new dhparam.pem is in place, nginx will be reloaded.
runtime: failed to create new OS thread (have 2 already; errno=22)
fatal error: newosproc

runtime stack:
runtime.throw(0x884500, 0x9)
	/usr/local/go1.6/src/runtime/panic.go:530 +0x90
runtime.newosproc(0xc820026000, 0xc820035fc0)
	/usr/local/go1.6/src/runtime/os1_linux.go:149 +0x18c
runtime.newm(0x932358, 0x0)
	/usr/local/go1.6/src/runtime/proc.go:1513 +0x135
runtime.main.func1()
	/usr/local/go1.6/src/runtime/proc.go:125 +0x2c
runtime.systemstack(0xa8e800)
	/usr/local/go1.6/src/runtime/asm_amd64.s:291 +0x79
runtime.mstart()
	/usr/local/go1.6/src/runtime/proc.go:1048

goroutine 1 [running]:
runtime.systemstack_switch()
	/usr/local/go1.6/src/runtime/asm_amd64.s:245 fp=0xc820020770 sp=0xc820020768
runtime.main()
	/usr/local/go1.6/src/runtime/proc.go:126 +0x62 fp=0xc8200207c0 sp=0xc820020770
runtime.goexit()
	/usr/local/go1.6/src/runtime/asm_amd64.s:1998 +0x1 fp=0xc8200207c8 sp=0xc8200207c0
```
This is a log from container.
```
  nginx-proxy:
    image: jwilder/nginx-proxy
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - ./ssl:/etc/nginx/certs
      - ./nginx_custom.conf:/etc/nginx/conf.d/nginx_custom.conf:ro
      - /var/run/docker.sock:/tmp/docker.sock:ro
```
And this is a `docker-compose.yml` code.

Anyone has any idea? Please let me know.

**Comment by Pintouch** [comment]
Hi,

Got the same on my M1 MBP. 

I'm using `docker engine 20.10.0` / `Compose 1.27.4`

And here is the stack trace :

```
WARNING: /etc/nginx/dhparam/dhparam.pem was not found. A pre-generated dhparam.pem will be used for now while a new one

is being generated in the background. Once the new dhparam.pem is in place, nginx will be reloaded.

runtime: failed to create new OS thread (have 2 already; errno=22)

fatal error: newosproc


runtime stack:

runtime.throw(0x884500, 0x9)

/usr/local/go1.6/src/runtime/panic.go:530 +0x90

runtime.newosproc(0xc820026000, 0xc820035fc0)

/usr/local/go1.6/src/runtime/os1_linux.go:149 +0x18c

runtime.newm(0x932358, 0x0)

/usr/local/go1.6/src/runtime/proc.go:1513 +0x135

runtime.main.func1()

/usr/local/go1.6/src/runtime/proc.go:125 +0x2c

runtime.systemstack(0xa8e800)

/usr/local/go1.6/src/runtime/asm_amd64.s:291 +0x79

runtime.mstart()

/usr/local/go1.6/src/runtime/proc.go:1048


goroutine 1 [running]:

runtime.systemstack_switch()

/usr/local/go1.6/src/runtime/asm_amd64.s:245 fp=0xc820020770 sp=0xc820020768

runtime.main()

/usr/local/go1.6/src/runtime/proc.go:126 +0x62 fp=0xc8200207c0 sp=0xc820020770

runtime.goexit()

/usr/local/go1.6/src/runtime/asm_amd64.s:1998 +0x1 fp=0xc8200207c8 sp=0xc8200207c0
```

**Comment by ninanung** [comment]
I saw some logs from container and found the `CMD ["forego", "start", "-r"]` command in [Dockerfile](https://github.com/nginx-proxy/nginx-proxy/blob/master/Dockerfile#L37) not working properly. This issue may related with [forego](https://github.com/jwilder/forego) and it is not a good news.

**Comment by bachhuberdesign** [comment]
Looks like there is an open PR that would address this issue: #1470

**Comment by ninanung** [comment]
@bachhuberdesign Thank you for the link! I will go test that commit.

**Comment by berndschimmer** [comment]
Hi, any news here? is it correct to change:
`jwilder/nginx-proxy:alpine`
to:
`jwilder/nginx-proxy:alpine@sha256:87a74118b4dca43bd280cae892ee8afc3c19c1d6`?
Any help is appreciated.

**Comment by ninanung** [comment]
Hi porks
I think this repository is not managed well. This [PR](https://github.com/nginx-proxy/nginx-proxy/pull/1470) opened few days ago, but not merged yet.
So i forked from PR owner's [repo](https://github.com/KWARC/nginx-proxy/tree/build-from-scratch) and tested it.
It kinda work for me. Container is running, but there may be side effects.

I copied his repo and built a image(link [here](https://hub.docker.com/r/ninanung/nginx-proxy)). Can test it if you want to.
I'm still on it, so let me know if someone solve this issue.

**Comment by berndschimmer** [comment]
Hi ninahung - your fork is working! Thx!

Does anybody now have also the problem that the 
```
volumes:
      - /Volumes/USB-HDD/:/data
```
are bound to 
`/host_mnt/Volumnes/USB-HDD` and can't be used?

**Comment by severin-bruhat** [comment]
Hello, 
Any update on this one? I can't make nginx-proxy work on my new macbook pro (M1)
Thank you

**Comment by berndschimmer** [comment]
Use https://hub.docker.com/r/ninanung/nginx-proxy from ninahung - this works perfectly. And my other issue with the wrongly bound volumes was also solved with the latest docker tech preview.

**Comment by ivtod** [comment]
@berndschimmer how are you using @ninanung fork? same config as jwilder/nginx-proxy? something like this is not working for me

```
services:
    web:
        image: php:7.3-apache
        ports:
          - "80:80"
          - "443:443"
        restart: always
        volumes:
            - ./:/var/www/html
        VIRTUAL_HOST: website.localhost
    api:
        image: php:7.3-apache
        ports:
          - "80:80"
          - "443:443"
        restart: always
        volumes:
            - ./:/var/www/html
        VIRTUAL_HOST: api.localhost
    nginx-proxy:
      image: ninanung/nginx-proxy
      ports:
        - "80:80"
        - "443:443"
      volumes:
        - /var/run/docker.sock:/tmp/docker.sock:ro
```

**Comment by berndschimmer** [comment]
here is my proxy configuration
```
 proxy:
    build: ./proxy
    restart: always
    ports:
      - 80:80
      - 443:443
    labels:
      - "com.github.jrcs.letsencrypt_nginx_proxy_companion.nginx_proxy"
    volumes:
      - certs:/etc/nginx/certs:ro
      - vhost.d:/etc/nginx/vhost.d
      - html:/usr/share/nginx/html
      - /var/run/docker.sock:/tmp/docker.sock:ro
    networks:
      - proxy-tier
```
and the second file (./proxy/dockerfile) contains 
```
FROM ninanung/nginx-proxy
```
I'm not sure if you need the second file

**Comment by tkw1536** [comment]
As of https://github.com/nginx-proxy/nginx-proxy/commit/d120ce9fdebf09a6c7b90ecff7eda0b7d2760ef3, there are now arm64 builds on DockerHub for the `nginxproxy/nginx-proxy` image. 

See [latest](https://hub.docker.com/layers/nginxproxy/nginx-proxy/latest/images/sha256-196f9a5fbeb1cd73745ed64fe81d430e776055c94869a8408342f681e515927b?context=explore) and [alpine](https://hub.docker.com/layers/nginxproxy/nginx-proxy/alpine/images/sha256-b3f15683324e5f420a206d21073a70922d9bb64213f89c935db6c05f091a2b0c?context=explore).  

/cc @buchdag This issue can probably be closed.

**Comment by buchdag** [comment]
As @tkw1536  mentioned, this has been fixed by #1470 + 4a1d5bfc58af35f5e90dcefb47b35a274c903150 + d120ce9fdebf09a6c7b90ecff7eda0b7d2760ef3

`arm64` and `arm/v7` images are availables from `jwilder/nginx-proxy` or the new registry `nginxproxy/nginx-proxy`
